### ISSUE TITLE
Texts.of() nests when unnecessary

### ISSUE THREAD

**Issue by kashike** [SpongePowered/SpongeAPI#899]
```java
        TextBuilder builder = Texts.builder();
        builder.color(TextColors.GRAY);
        builder.append(Texts.of(TextColors.DARK_AQUA, "gratimax"), Texts.of(" is love, "), Texts.of(TextColors.DARK_AQUA, "gratimax"), Texts.of(" is life"));
        this.logger.info(Texts.json().to(builder.build()));
```
Current output:
```json
{
    "color": "gray",
    "extra": [
        {
            "extra": [
                {
                    "color": "dark_aqua",
                    "text": "gratimax"
                }
            ],
            "text": ""
        },
        " is love, ",
        {
            "extra": [
                {
                    "color": "dark_aqua",
                    "text": "gratimax"
                }
            ],
            "text": ""
        },
        " is life"
    ],
    "text": ""
}
```

Desired output:
```json
{
    "color": "gray",
    "extra": [
        {
            "color": "dark_aqua",
            "text": "gratimax"
        },
        " is love, ",
        {
            "color": "dark_aqua",
            "text": "gratimax"
        },
        " is life"
    ],
    "text": ""
}
```
____
The following results in the desired output, but cannot be used in many places.
```java
        this.logger.info(Texts.json().to(Texts.of(TextColors.DARK_AQUA, "gratimax", TextColors.GRAY, " is love ", TextColors.DARK_AQUA, "gratimax", TextColors.GRAY, " is life")));
```
```json
{
    "extra": [
        {
            "color": "dark_aqua",
            "text": "gratimax"
        },
        {
            "color": "gray",
            "text": " is love "
        },
        {
            "color": "dark_aqua",
            "text": "gratimax"
        },
        {
            "color": "gray",
            "text": " is life"
        }
    ],
    "text": ""
}
```

**Comment by ST-DDT** [comment]
Can you tell me why the second one cannot be used?

PS my Text API PR fixes this single element issue. #808

**Comment by kashike** [comment]
@ST-DDT The second one is not a TextBuilder, which is needed to construct certain messages.

**Comment by Minecrell** [comment]
`Texts.of(...)` is not really designed to design nice output (why would you ever read it yourself anyway), but rather only a simple way to send a message. I'm not sure if it would be worth optimizing it in a way that would make `Texts.of(...)` much more complicated when it's normally just designed to be an utility method.

**Comment by JBYoshi** [comment]
Partially fixed in c452da63263cb7b948d4235fced67422ddea954a.
